### DISCUSSION TITLE
mehrere lesende GAs

### DISCUSSION THREAD

**Discussion by tobihoby** [brutella/hkknx-public#219] [Q&A]
Hallo,

wie kann ich mehrere lesende GAs bei zum Beispiel einem Schalter eintragen?

Gruß
Tobi

**Comment by brutella** [comment]
Bitte erklär deinen Anwendungsfall etwas genauer. 
Warum sollte der Schalter mehrere GAs verwenden?

**Comment by tobihoby** [comment]
hat sich erledigt, Denkfehler...

**Comment by frankschneider** [comment]
Ich habe die selbe Frage, vielleicht aber auch einen Denkfehler? 

Ich habe bei allen Leuchten mehrere Gruppenadressen. 1. Leuchte direkt, 2. Alle Leuchten im Raum, 3. Alle Leuchten im Stockwerk, 4. Alle Leuchten im Gesamten Haus.

Wenn ich nun die Leuchten direkt über den KNX Bus z.B. im Gesamten Stockwerk ausschalte, bekommt die hkknx Brücke das nicht mit und denkt, die Leuchte ist noch an. Versuche ich sie dann später über die Brücke einzuschalten, passiert erstmal nichts. Ich muss die Leuchte dann zuerst aus- und dann wieder einschalten, damit was passiert. Was mache ich hier falsch? Im Gira Homeserver gibt es dazu z.B. mehrere lesende "Watchadressen", die mithören.

Vielen Dank im Voraus für eure Hilfe.

**Reply by chef76** [reply]
Im Aktor müssen die Status GA's lesbar sein. Dann funktioniert das. Kann sein, dass der X1 das übernehmen kann, aber eigentlich braucht HKKNX den X1 nicht. Geh mal in der ETS am Aktor auf den Status vom Licht -> Eigenschaften und prüfe ob Lesen angehakt ist.

Wenn du Homekit öffnest fragt er alle Geräte an und wenn die den Status nicht zurücksenden, hast du genau das beschriebene Verhalten.

HKKNX speichert keine Zustände zwischen und kriegt damit das Schalten nicht mit, wenn Homekit nicht offen war.

Die Lesen GAs, muss natürlich auch in HKKNX gepflegt sein.

**Reply by brutella** [reply]
> Wenn du Homekit öffnest fragt er alle Geräte an und wenn die den Status nicht zurücksenden, hast du genau das beschriebene Verhalten.

Das stimmt so nicht. Beim Starten von hkknx werden alle Geräte initialisiert, dh die Zustände werden abgefragt. Ab dann wird auf die Lesen-GAs gehorcht und die internen Zustände aktualisiert.

> HKKNX speichert keine Zustände zwischen und kriegt damit das Schalten nicht mit, wenn Homekit nicht offen war.

Doch. hkknx speichert die Zustände, da ansonsten die Abfrage in der Apple Home-App zu lange dauern würde.
